### PR TITLE
mbank-cli: new-command: list-profiles

### PR DESCRIPTION
**Pull request by adamgolebiowski**
new command to list business/personal profiles

### CODE REVIEW

File: mbank-cli [hunk 1]
```diff
@@ -1879,6 +1879,35 @@ sub do_notices
...
+    scalar(@args) > 0
+        or $profile_type = 'all';
+    scalar(@args) <= 1
+        or user_error("list-profiles: too many arguments");
+    my @valid_types = qw(personal business all);
+    grep(/^$profile_type/, @valid_types)
+        or user_error("list-profiles: unknown profile type $profile_type");
+    my %profiles = %{$login_info->{profiles}};
+    foreach my $account_name (keys %profiles) {
+        my @account_data = $profiles{$account_name};
```
[the remark this answers is not in the document]
Comment (adamgolebiowski) [reply mbank-cli]: you are right - I will look into it (I am not a perl guy, so...)

File: mbank-cli [hunk 2]
```diff
@@ -1879,6 +1879,35 @@ sub do_notices
...
+        // internal_error('missing login info');
+    my @args = @{delete($options{args})};
+    check_for_unexpected_options(%options);
+    my ($profile_type) = @args;
+    scalar(@args) > 0
+        or $profile_type = 'all';
+    scalar(@args) <= 1
+        or user_error("list-profiles: too many arguments");
+    my @valid_types = qw(personal business all);
+    grep(/^$profile_type/, @valid_types)
```
[the remark this answers is not in the document]
Comment (adamgolebiowski) [reply mbank-cli]: indeed `mbank-cli list-profiles p` prints nothing, will work on this

File: mbank-cli [hunk 3]
```diff
@@ -1879,6 +1879,35 @@ sub do_notices
...
+        or user_error("list-profiles: unknown profile type $profile_type");
+    my %profiles = %{$login_info->{profiles}};
+    foreach my $account_name (keys %profiles) {
+        my @account_data = $profiles{$account_name};
+        my $active_str = '(inactive)';
+        $active_str = "active" if defined($account_data[0][0]);
```
[the remark this answers is not in the document]
Comment (adamgolebiowski) [reply mbank-cli]: works as intended, here's my case (names anonymized):
```
[adamg@adamg mbank-cli]$ ./mbank-cli list-profiles
Company1      active
personal        (inactive)
Company2        active
[adamg@adamg mbank-cli]$ ./mbank-cli list-profiles
personal        (inactive)
Company1      active
Company2        active
[adamg@adamg mbank-cli]$

```

I think we should also sort the output (personal first, the business alphabetically)

File: mbank-cli [hunk 4]
```diff
@@ -1879,6 +1879,35 @@ sub do_notices
...
+        if ( $profile_type eq 'all'
+                or ($profile_type eq 'business' and $account_name ne 'personal')
+                or ($profile_type eq 'personal' and $account_name eq 'personal')
+        ) {
```
[the remark this answers is not in the document]
Comment (adamgolebiowski) [reply mbank-cli]: something like this?
```
my $is_all = ($profile_type eq 'all');
my $is_business = ($profile_type eq 'business' and $account_name ne 'personal');
my $is_personal = ($profile_type eq 'personal' and $account_name eq 'personal');
if ( $is_all or $is_business or $is_personal) {
   ...
}
```

File: mbank-cli (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamgolebiowski) [reply mbank-cli]: after this change mbank-cli fails with:
`Can't use string ("53304040") as an ARRAY ref while "strict refs" in use at ./mbank-cli line 1901.`
will look into it

File: mbank-cli (hunk 3 above)
[the remark this answers is not in the document]
Comment (jwilk) [reply mbank-cli]: This shows two profiles as active. Unless I completely misunderstood how profiles work, only one of them can be active at one time.

The information which profile is active, is available in the HTML; for example, a user in #58 had:
```js
Ebre.Venezia.ProfileData = {"iProfiles":{"appURL":"#market/group/IO","active":true,"profiles":[{"profileCode":"T","active":true,"previouslyActive":false,"firmName":null}]},"fProfiles":{"appURL":"#market/group/BO","active":false,"profiles":[{"profileCode":"11581694","active":false,"previouslyActive":true,"firmName":" COMPANY NAME"}]}};
```
But this information is not currently gathered by `_extract_login_profiles`.

File: mbank-cli (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwilk) [reply mbank-cli]: `@account_data` is now less-nested, so you need to drop one `[0]` from `$account_data[0][0]`.

File: mbank-cli (hunk 4 above)
[the remark this answers is not in the document]
Comment (jwilk) [reply mbank-cli]: Right, that's what I had in mind.
But please remove space after `(`.
You can also drop parentheses in the first line.

File: mbank-cli (hunk 3 above)
[the remark this answers is not in the document]
Comment (adamgolebiowski) [reply mbank-cli]: You are right, the confusion comes from different definition of "active" profile that I used. 
In mbank-cli, active profile is the one you switched to, what I meant is more what you would call "defined" or "enabled" - e.g. something you signed contract for. 

That said, what about chaging the output of list-profiles to (subject to tab-formatting):
```
[adamg@adamg mbank-cli]$ ./mbank-cli list-profiles
Company1      enabled      active
personal        disabled      
Company2        enabled
[adamg@adamg mbank-cli]$
```
On this login I have two business profiles and did not sign contract for personal one and the one for Company1 is currently active (mbank-cli activate-profile).

Here's another example:
```
[adamg@adamg mbank-cli]$ ./mbank-cli list-profiles
business      disabled
personal        enabled      active
[adamg@adamg mbank-cli]$
```

On this login I have only personal profile which is enabled.

**Pull request closed without merging**